src/vs/platform/storage/common/storage.ts

```diff
@@ -140,6 +140,11 @@ export interface IStorageService {
 	 */
 	log(): void;
 
+	/**
+	 * Returns the profile used for the profile storage
+	 */
+	getProfileStorageProfile(): IUserDataProfile | undefined;
```

Comment:
This looks like a hack to me because I think only some storage services will have a profile and not all of them. And the name is strange too having 2x profile in its name. Please see if another solution is possible and then I would prefer just `.profile`.

Reply:
Hmm, may be this is an hack/workaround - but this will reduce traffic between shared process and main when there are no profiles. `IProfileStorageService` checks if `IStorageService` is handling the given profile and read/update the storage data through storage service instead of opening a new storage client connection to main.